File: UsingParceler/app/src/main/java/github/nisrulz/sample/usingparceler/Person.java
package github.nisrulz.sample.usingparceler;

import org.parceler.Parcel;

import static org.parceler.Parcel.Serialization;

@Parcel(Serialization.BEAN)
class Person {
  private String name;
  private String address;
  private int age;
  private String occupation;

  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public String getAddress() {
    return address;
  }

  public void setAddress(String address) {
    this.address = address;
  }

  public int getAge() {
    return age;
  }

  public void setAge(int age) {
    this.age = age;
  }

  public String getOccupation() {
    return occupation;
  }

  public void setOccupation(String occupation) {
    this.occupation = occupation;
  }

  @Override
  public String toString() {
    StringBuilder data = new StringBuilder();
    data.append("Name : ")
        .append(name)
        .append("\n")
        .append("Address : ")
        .append(address)
        .append("\n")
        .append("Age : ")
        .append(age)
        .append("\n")
        .append("Occupation : ")
        .append(occupation);
    return data.toString();
  }
}

File: UsingParcelize/app/src/main/java/github/nisrulz/usingparcelize/Person.kt
package github.nisrulz.usingparcelize

import android.os.Parcelable
import kotlinx.android.parcel.Parcelize

@Parcelize
data class Person(
    val name: String = "",
    val address: String = "",
    val age: Int = 0,
    val occupation: String = ""
) : Parcelable {

    override fun toString(): String {
        return "Name: $name\nAge: $age\nAddress: $address\nOccupation: $occupation".toString()
    }
}